roles/conjur_host_identity/tasks/main.yml

```diff
@@ -1,4 +1,5 @@
 ---
+- include: clean_identity_config.yml
```

Comment:
Using this implementation paints the users into a bit of a corner - the only way cleanup will be triggered is if they:
1. Provide a host with an identity
2. Run some task on the managed host
3. Revoke the last identity, and provide the same host with another identity

These identities are more long-term than that - we should be able to:
1. Provide a host with an identity
2. Run a task
3. Run another task
4. ...
5. Revoke the identity

This [StackExchange answer](https://serverfault.com/a/612125) offers a solution for giving an Ansible role a cleanup method triggered by setting a `state` variable to `absent` - that might be a better path for our purposes. Then a user would be able to grant an identity, and tear it down when required.

Reply:
I have been working on it today , I would like to talk on it's few aspects .